File: Rehla-integration/src/main/java/models/Preference.java
package models;

/**
 * Entity matching the database table `preference`.
 * Links to a user via personne_id.
 */
public class Preference {

    private int id;
    private Double budgetMin;
    private Double budgetMax;
    private String typesVoyage;
    private String centresInteret;
    private int personneId;

    public Preference() {}

    public Preference(int id, Double budgetMin, Double budgetMax, String typesVoyage,
                      String centresInteret, int personneId) {
        this.id = id;
        this.budgetMin = budgetMin;
        this.budgetMax = budgetMax;
        this.typesVoyage = typesVoyage;
        this.centresInteret = centresInteret;
        this.personneId = personneId;
    }

    public Preference(Double budgetMin, Double budgetMax, String typesVoyage,
                      String centresInteret, int personneId) {
        this.budgetMin = budgetMin;
        this.budgetMax = budgetMax;
        this.typesVoyage = typesVoyage;
        this.centresInteret = centresInteret;
        this.personneId = personneId;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public Double getBudgetMin() { return budgetMin; }
    public void setBudgetMin(Double budgetMin) { this.budgetMin = budgetMin; }

    public Double getBudgetMax() { return budgetMax; }
    public void setBudgetMax(Double budgetMax) { this.budgetMax = budgetMax; }

    public String getTypesVoyage() { return typesVoyage; }
    public void setTypesVoyage(String typesVoyage) { this.typesVoyage = typesVoyage; }

    public String getCentresInteret() { return centresInteret; }
    public void setCentresInteret(String centresInteret) { this.centresInteret = centresInteret; }

    public int getPersonneId() { return personneId; }
    public void setPersonneId(int personneId) { this.personneId = personneId; }
}

File: src/Entity/Preference.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

use App\Entity\Personne;

#[ORM\Entity]
class Preference
{

    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: "integer")]
    private ?int $id = null;

    #[ORM\Column(name: "budgetMin", type: "float", nullable: true)]
    private ?float $budgetMin = null;

    #[ORM\Column(name: "budgetMax", type: "float", nullable: true)]
    private ?float $budgetMax = null;

    #[ORM\Column(name: "typesVoyage", type: "text", nullable: true)]
    private ?string $typesVoyage = null;

    #[ORM\Column(name: "centresInteret", type: "text", nullable: true)]
    private ?string $centresInteret = null;

    #[ORM\ManyToOne(targetEntity: Personne::class, inversedBy: "preferences")]
    #[ORM\JoinColumn(name: 'personne_id', referencedColumnName: 'id', onDelete: 'CASCADE', nullable: true)]
    private ?Personne $personne_id = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getBudgetMin(): ?float
    {
        return $this->budgetMin;
    }

    public function setBudgetMin(?float $value): void
    {
        $this->budgetMin = $value;
    }

    public function getBudgetMax(): ?float
    {
        return $this->budgetMax;
    }

    public function setBudgetMax(?float $value): void
    {
        $this->budgetMax = $value;
    }

    public function getTypesVoyage(): ?string
    {
        return $this->typesVoyage;
    }

    public function setTypesVoyage(?string $value): void
    {
        $this->typesVoyage = $value;
    }

    public function getCentresInteret(): ?string
    {
        return $this->centresInteret;
    }

    public function setCentresInteret(?string $value): void
    {
        $this->centresInteret = $value;
    }

    public function getPersonne_id(): ?Personne
    {
        return $this->personne_id;
    }

    public function setPersonne_id(?Personne $value): void
    {
        $this->personne_id = $value;
    }
}
